src/lib.rs

```diff
@@ -94,36 +45,51 @@ pub fn schedule_notification(
 ///
 /// ```no_run
 /// # use mac_notification_sys::*;
-/// // daliver a silent notification
-/// let _ = send_notification("Title", &None, "This is the body", &None).unwrap();
+/// // deliver a silent notification
+/// let _ = send_notification("Title", None, "This is the body", None).unwrap();
 /// ```
 pub fn send_notification(
     title: &str,
-    subtitle: &Option<&str>,
+    subtitle: Option<&str>,
     message: &str,
-    sound: &Option<&str>,
-) -> NotificationResult<()> {
-    let use_sound = match sound {
-        Some(sound) if check_sound(sound) => sound,
-        _ => "_mute",
+    options: Option<&NotificationOptions>,
+) -> NotificationResult<NotificationResponse> {
+    if let Some(options) = &options {
+        if let Some((delivery_date, _)) = options.delivery_date {
+            ensure!(
+                delivery_date >= Utc::now().timestamp() as f64,
+                NotificationError::ScheduleInThePast
+            );
+        }
     };
 
+    let options = options
+        .unwrap_or(&NotificationOptions::new())
```

Comment:
if we have a default we could simply use `unwrap_or_default()`

Reply:
After adding the required changes to allow for default, I can't seem to `unwrap_or_default()` an option of reference.